File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Par_ProfiloOrario/Models/par-profilo-orario-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";
import { Par_OrarioIntervalloHHModel } from "../../Par_OrarioIntervalloHH/Models/par-orario-intervallo-hh-model";
import { Par_ProfiloOrarioGGModel } from "../../Par_ProfiloOrarioGG/Models/par-profilo-orario-gg-model";

export class Par_ProfiloOrarioModel {
  id: number;
  idAz_Anagrafica: number;
  codice: string;
  descrizione: string;
  numGiorniCiclo: number;
  tipoProfilo: number; // 0= settimanale , 1= ciclico

  constructor() {
    this.id = 0;
    this.idAz_Anagrafica = 0;
    this.codice = '';
    this.descrizione = '';
    this.numGiorniCiclo = 0;
    this.tipoProfilo = 0;
  }

}

export class Par_ProfiloOrario_GetAllInModel { }
export class Par_ProfiloOrario_GetAllOutModel extends ModelResult {
  par_ProfiloOrario: Par_ProfiloOrarioModel[];
  constructor() {
    super();
    this.par_ProfiloOrario = [];
  }
}

export class Par_ProfiloOrario_GetInModel {
  id: number;
}
export class Par_ProfiloOrario_GetOutModel extends ModelResult {
  par_ProfiloOrario: Par_ProfiloOrarioModel;
  par_ProfiloOrarioGG: Par_ProfiloOrarioGGModel[];
  par_OrarioIntervalloHH: Par_OrarioIntervalloHHModel[];
  constructor() {
    super();
    this.par_ProfiloOrario = new Par_ProfiloOrarioModel();
    this.par_ProfiloOrarioGG = [];
  }
}

export class Par_ProfiloOrario_PutInModel {
  par_ProfiloOrario: Par_ProfiloOrarioModel;
  par_ProfiloOrarioGG: Par_ProfiloOrarioGGModel[];
  constructor() {
    this.par_ProfiloOrario = new Par_ProfiloOrarioModel();
    this.par_ProfiloOrarioGG = [];
  }
}
export class Par_ProfiloOrario_PutOutModel extends ModelResult {
  par_ProfiloOrario: Par_ProfiloOrarioModel;
  par_ProfiloOrarioGG: Par_ProfiloOrarioGGModel[];
  constructor() {
    super();
    this.par_ProfiloOrario = new Par_ProfiloOrarioModel();
    this.par_ProfiloOrarioGG = [];
  }
}

export class Par_ProfiloOrario_DeleteInModel {
  id: number;
}
export class Par_ProfiloOrario_DeleteOutModel extends ModelResult { }

File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Par_ProfiloOrarioService/Models/Par_ProfiloOrarioModel.cs
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Az_SubCommessaService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_OrarioIntervalloHHService.Models;
using nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_ProfiloOrarioGGService.Models;
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_ProfiloOrarioService.Models
{
    public class Par_ProfiloOrarioModel
    {
        public int Id { get; set; }
        public int IdAz_Anagrafica { get; set; }
        public string Codice { get; set; } = string.Empty;
        public string Descrizione { get; set; } = string.Empty;
        public int NumGiorniCiclo { get; set; }
        public int TipoProfilo { get; set; }
        public int idPar_Orario_Festivo { get; set; }
        
    }

    public class Par_ProfiloOrario_GetAllInModel { }
    public class Par_ProfiloOrario_GetAllOutModel : ModelResult
    {
        public List<Par_ProfiloOrarioModel> Par_ProfiloOrario { get; set; } = new List<Par_ProfiloOrarioModel>();
    }

    public class Par_ProfiloOrario_GetInModel
    {
        public int Id { get; set; }
    }
    public class Par_ProfiloOrario_GetOutModel : ModelResult
    {
        public Par_ProfiloOrarioModel? Par_ProfiloOrario { get; set; }
        public List<Par_ProfiloOrarioGGModel> Par_ProfiloOrarioGG { get; set; } = new List<Par_ProfiloOrarioGGModel>();
        public List<Par_OrarioIntervalloHHModel> Par_OrarioIntervalloHH { get; set; } = new List<Par_OrarioIntervalloHHModel>();

    }

    public class Par_ProfiloOrario_PutInModel
    {
        public Par_ProfiloOrarioModel Par_ProfiloOrario { get; set; } = new Par_ProfiloOrarioModel();
        public List<Par_ProfiloOrarioGGModel> Par_ProfiloOrarioGG { get; set; } = new List<Par_ProfiloOrarioGGModel>();
    }
    public class Par_ProfiloOrario_PutOutModel : ModelResult
    {
        public Par_ProfiloOrarioModel? Par_ProfiloOrario { get; set; }
        public List<Par_ProfiloOrarioGGModel> Par_ProfiloOrarioGG { get; set; } = new List<Par_ProfiloOrarioGGModel>();
    }

    public class Par_ProfiloOrario_DeleteInModel
    {
        public int Id { get; set; }
    }
    public class Par_ProfiloOrario_DeleteOutModel : ModelResult { }
}
